Error: Invariant: Method expects to have requestAsyncStorage, none available

Hey, I cloned the repo, followed the readme, but i see this error? Any inputs ?

I have already searched online without any luck

Also, is it mandatory to start the supabase local stack ?

```
Error: Invariant: Method expects to have requestAsyncStorage, none available
    at Object.cookies (webpack-internal:///(sc_server)/./node_modules/.pnpm/next@13.4.7-canary.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/client/components/headers.js:46:15)
    at NextRouteHandlerAuthStorageAdapter.getCookie (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+auth-helpers-nextjs@0.7.2_@supabase+supabase-js@2.26.0/node_modules/@supabase/auth-helpers-nextjs/dist/index.js:230:42)
    at NextRouteHandlerAuthStorageAdapter.getItem (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+auth-helpers-shared@0.4.1_@supabase+supabase-js@2.26.0/node_modules/@supabase/auth-helpers-shared/dist/index.js:261:28)
    at eval (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:151:37)
    at Generator.next (<anonymous>)
    at eval (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:59:71)
    at new Promise (<anonymous>)
    at __awaiter (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:41:12)
    at getItemAsync (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:150:38)
    at SupabaseAuthClient.eval (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/GoTrueClient.js:1356:73)
Error: Invariant: Method expects to have requestAsyncStorage, none available
    at Object.cookies (webpack-internal:///(sc_server)/./node_modules/.pnpm/next@13.4.7-canary.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/client/components/headers.js:46:15)
    at NextRouteHandlerAuthStorageAdapter.getCookie (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+auth-helpers-nextjs@0.7.2_@supabase+supabase-js@2.26.0/node_modules/@supabase/auth-helpers-nextjs/dist/index.js:230:42)
    at NextRouteHandlerAuthStorageAdapter.getItem (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+auth-helpers-shared@0.4.1_@supabase+supabase-js@2.26.0/node_modules/@supabase/auth-helpers-shared/dist/index.js:261:28)
    at eval (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:151:37)
    at Generator.next (<anonymous>)
    at eval (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:59:71)
    at new Promise (<anonymous>)
    at __awaiter (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:41:12)
    at getItemAsync (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:150:38)
    at SupabaseAuthClient.eval (webpack-internal:///(sc_server)/./node_modules/.pnpm/@supabase+gotrue-js@2.39.0/node_modules/@supabase/gotrue-js/dist/main/GoTrueClient.js:793:75)
```

fresh clone, npm install, npm run dev:
```
npm run dev

> @supabase-community/vercel-ai-chatbot@0.0.2 dev
> next dev

- ready started server on 0.0.0.0:3000, url: http://localhost:3000
- info Loaded env from /Users/lukaszgandecki/projects/vercel-test/vercel-chatbot-fork/.env
- warn You have enabled experimental feature (serverActions) in next.config.js.
- warn Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.

(node:27376) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
- event compiled client and server successfully in 130 ms (20 modules)
- wait compiling...
- event compiled client and server successfully in 77 ms (20 modules)
- wait compiling /middleware (client and server)...
- event compiled successfully in 329 ms (179 modules)
- wait compiling /sign-in/page (client and server)...
- warn ./node_modules/node-fetch/lib/index.js
Module not found: Can't resolve 'encoding' in '/Users/lukaszgandecki/projects/vercel-test/vercel-chatbot-fork/node_modules/node-fetch/lib'

Import trace for requested module:
./node_modules/node-fetch/lib/index.js
./node_modules/cross-fetch/dist/node-ponyfill.js
./node_modules/@supabase/supabase-js/dist/main/lib/fetch.js
./node_modules/@supabase/supabase-js/dist/main/SupabaseClient.js
./node_modules/@supabase/supabase-js/dist/main/index.js
./node_modules/@supabase/auth-helpers-shared/dist/index.js
./node_modules/@supabase/auth-helpers-nextjs/dist/index.js
./auth.ts
./app/sign-in/page.tsx

./node_modules/node-fetch/lib/index.js
Module not found: Can't resolve 'encoding' in '/Users/lukaszgandecki/projects/vercel-test/vercel-chatbot-fork/node_modules/node-fetch/lib'

Import trace for requested module:
./node_modules/node-fetch/lib/index.js
./node_modules/cross-fetch/dist/node-ponyfill.js
./node_modules/@supabase/supabase-js/dist/main/lib/fetch.js
./node_modules/@supabase/supabase-js/dist/main/SupabaseClient.js
./node_modules/@supabase/supabase-js/dist/main/index.js
./node_modules/@supabase/auth-helpers-shared/dist/index.js
./node_modules/@supabase/auth-helpers-nextjs/dist/index.js
./components/login-button.tsx
Error: Invariant: Method expects to have requestAsyncStorage, none available
    at Object.cookies (webpack-internal:///(sc_server)/./node_modules/next/dist/client/components/headers.js:46:15)
    at NextRouteHandlerAuthStorageAdapter.getCookie (webpack-internal:///(sc_server)/./node_modules/@supabase/auth-helpers-nextjs/dist/index.js:230:42)
    at NextRouteHandlerAuthStorageAdapter.getItem (webpack-internal:///(sc_server)/./node_modules/@supabase/auth-helpers-shared/dist/index.js:261:28)
    at eval (webpack-internal:///(sc_server)/./node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:151:37)
    at Generator.next (<anonymous>)
    at eval (webpack-internal:///(sc_server)/./node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:59:71)
    at new Promise (<anonymous>)
    at __awaiter (webpack-internal:///(sc_server)/./node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:41:12)
    at getItemAsync (webpack-internal:///(sc_server)/./node_modules/@supabase/gotrue-js/dist/main/lib/helpers.js:150:38)
    at SupabaseAuthClient.eval (webpack-internal:///(sc_server)/./node_modules/@supabase/gotrue-js/dist/main/GoTrueClient.js:1357:73)

```

--- a/app/api/chat/route.ts
+++ b/app/api/chat/route.ts
@@ -8,7 +8,7 @@ import { Database } from '@/lib/db_types'
 import { auth } from '@/auth'
 import { nanoid } from '@/lib/utils'
 
-export const runtime = 'edge'
+export const runtime = 'nodejs'
 
 const configuration = new Configuration({
   apiKey: process.env.OPENAI_API_KEY

--- a/app/chat/[id]/page.tsx
+++ b/app/chat/[id]/page.tsx
@@ -5,7 +5,7 @@ import { auth } from '@/auth'
 import { getChat } from '@/app/actions'
 import { Chat } from '@/components/chat'
 
-export const runtime = 'edge'
+export const runtime = 'nodejs'
 export const preferredRegion = 'home'
 
 export interface ChatPageProps {

--- a/app/page.tsx
+++ b/app/page.tsx
@@ -1,7 +1,7 @@
 import { nanoid } from '@/lib/utils'
 import { Chat } from '@/components/chat'
 
-export const runtime = 'edge'
+export const runtime = 'nodejs'
 
 export default function IndexPage() {
   const id = nanoid()

--- a/app/share/[id]/page.tsx
+++ b/app/share/[id]/page.tsx
@@ -6,7 +6,7 @@ import { getSharedChat } from '@/app/actions'
 import { ChatList } from '@/components/chat-list'
 import { FooterText } from '@/components/footer'
 
-export const runtime = 'edge'
+export const runtime = 'nodejs'
 export const preferredRegion = 'home'
 
 interface SharePageProps {

